simulator/risc_v/t/unit_test.cpp

```diff
@@ -599,6 +599,43 @@ TEST_CASE("RISCV RV64 gorc")
     }
 }
 
+TEST_CASE("RISCV RV32 gorci")
+{
+    CHECK( RISCVInstr<uint32>(0x28D6'5593).get_disasm() == "gorci $a1, $a2, 13");
+    CHECK( RISCVInstr<uint32>(0x28F6'5593).get_disasm() == "gorci $a1, $a2, 15");
+    CHECK( RISCVInstr<uint32>(0x29C6'5593).get_disasm() == "gorci $a1, $a2, 28");
+    std::vector<TestData<uint32>> cases = {
+            {0x1111'0000, 0x1, 0x3333'0000},
+            {0x0022'0033, 0x1, 0x00AA'00FF},
+            {0x0C0C'0F0F, 0x1, 0xCCCC'FFFF},
+            {0x8712'4789, 0x1, all_ones<uint32>()},
+            {1, 0x1, all_ones<uint32>()},
+            {0x1252'0391, 0x1, 0xFFFF'0FFF},
+    };
+	uint32 shamts[6] = {0x1, 0x2, 0x5, 0xC, 0xFF, 0x3};
```

Comment:
We do not use C-style arrays, consider using C++ structures instead: https://github.com/MIPT-ILab/mipt-mips/wiki/Cpp-guidelines#arrays